gh-141004: Document `PyFile_OpenCode` and `PyFile_OpenCodeObject` (#141413)
python/cpython: Doc/c-api/file.rst

   .. versionadded:: 3.8

.. c:function:: PyObject *PyFile_OpenCodeObject(PyObject *path)

>    Open *path* and read the text. *path* must be a Python :class:`str`

Reviewer: ```suggestion
   Open *path* with the mode ``'rb'``. *path* must be a Python :class:`str`
```
"text" is wrong, so I suggest we do like in [`io.open_code`](https://docs.python.org/3/library/io.html#io.open_code).

   object. The behavior of this function may be overridden by
   :c:func:`PyFile_SetOpenCodeHook` to allow for some preprocessing of the
   text.

>    This is analogous to :func:`io.open_code` in Python.

Reviewer: ```suggestion
   This is analogous to :func:`io.open_code` or ``open(path, 'rb')`` in Python.
```
Also ~~stolen~~ borrowed from the `io.open_code` docs.

Author: I don't think this is useful if we already state that it opens with `rb` above.

   .. versionadded:: 3.8

.. c:function:: PyObject *PyFile_OpenCodeObject(PyObject *path)

   Open *path* and read the text. *path* must be a Python :class:`str`
   object. The behavior of this function may be overridden by
   :c:func:`PyFile_SetOpenCodeHook` to allow for some preprocessing of the
   text.

   This is analogous to :func:`io.open_code` in Python.

   On success, this function returns a :term:`strong reference` to a Python
   file object. On failure, this function returns ``NULL`` with an exception
   set.

   .. versionadded:: 3.8

.. c:function:: PyObject *PyFile_OpenCode(const char *path)

   Similar to :c:func:`PyFile_OpenCodeObject`, but *path* is a
   :c:expr:`const char*` that is decoded by the

>    :term:`filesystem encoding and error handler`.

Reviewer: UTF-8 encoding, no (also implied by the parameter name)?

https://github.com/python/cpython/blob/d890aba748e5213585f9f906888999227dc3fa9c/Objects/fileobject.c#L528

Author: Ah, good catch.
